compiler/luci-interpreter/src/kernels/RmsNorm.cpp

```diff
@@ -0,0 +1,145 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/RmsNorm.h"
+
+#include "kernels/Utils.h"
+
+#include <tensorflow/lite/kernels/internal/common.h>
+#include <cmath>
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+RmsNorm::RmsNorm(const Tensor *input, const Tensor *gamma, const Tensor *beta, Tensor *output,
+                 const RmsNormParams &params)
+  : KernelWithParams<RmsNormParams>({input, gamma, beta}, {output}, params)
+{
+}
+
+void RmsNorm::configure()
+{
+  auto num_dims = input()->shape().num_dims();
+  LUCI_INTERPRETER_CHECK(num_dims == 3 || num_dims == 4);
+  LUCI_INTERPRETER_CHECK(input()->element_type() == output()->element_type());
+  LUCI_INTERPRETER_CHECK(gamma()->element_type() == input()->element_type());
+  LUCI_INTERPRETER_CHECK(beta()->element_type() == input()->element_type());
+  LUCI_INTERPRETER_CHECK(gamma()->shape().num_dims() == 1);
+  LUCI_INTERPRETER_CHECK(beta()->shape().num_dims() == 1);
+  LUCI_INTERPRETER_CHECK((gamma()->shape().dim(0) == input()->shape().dim(num_dims - 1)) ||
+                         (gamma()->shape().dim(0) == 1));
+  LUCI_INTERPRETER_CHECK((beta()->shape().dim(0) == input()->shape().dim(num_dims - 1)) ||
+                         (beta()->shape().dim(0) == 1));
+
+  output()->resize(input()->shape());
+}
+
+void RmsNorm::execute() const
+{
+  switch (input()->element_type())
+  {
+    case DataType::FLOAT32:
+      evalFloat();
+      break;
+    default:
+      throw std::runtime_error("luci-intp RmsNorm Unsupported type.");
+  }
+}
+
+void RmsNorm::evalFloat() const
+{
+  tflite::RuntimeShape input_shape = getTensorShape(input());
+  auto output_shape = getTensorShape(output());
+
+  const float *input_data = getTensorData<float>(input());
+  const float *gamma_data = getTensorData<float>(gamma());
+  auto gamma_shape = getTensorShape(gamma());
+  bool single_gamma = gamma_shape.DimensionsCount() == 1 && gamma_shape.Dims(0) == 1;
+  const float *beta_data = getTensorData<float>(beta());
+  auto beta_shape = getTensorShape(beta());
+  bool single_beta = beta_shape.DimensionsCount() == 1 && beta_shape.Dims(0) == 1;
+  float *output_data = getTensorData<float>(output());
+
+  if (input_shape.DimensionsCount() == 4)
+  {
+    // Dimensions for image case are (N x H x W x C)
+    const int32_t batches = tflite::MatchingDim(input_shape, 0, output_shape, 0);
+    const int32_t heights = tflite::MatchingDim(input_shape, 1, output_shape, 1);
+    const int32_t widths = tflite::MatchingDim(input_shape, 2, output_shape, 2);
+    const int32_t channels = tflite::MatchingDim(input_shape, 3, output_shape, 3);
```

Comment:
Q) can we use our implementation (not sure there exist any), not tflite codes?
using `tflite` codes in other nodes are, as I understand, to invoke `tflite` kernel codes but as I see kernel itself is implemented here so it would be better not to use `tflite` codes.

Reply:
> Q) can we use our implementation (not sure there exist any), not tflite codes? using `tflite` codes in other nodes are, as I understand, to invoke `tflite` kernel codes but as I see kernel itself is implemented here so it would be better not to use `tflite` codes.

In runtime, I could find an implementation (ex, compute/cker/include/Shape.h), but nothing in frontend.
MatchingDim() is used as a utility function in frontend codes(including instancenorm and etc)
It's short and simple, so I can define it, but it's not consistent with other frontend codes, I think.

Reply:
OK... maybe we can update this later.
Let's go as-is.